File: python/palindrome.py
class Solution(object):
    def isPalindrome(self, x):
        if (x < -9): return False
        if (x < 9 and x > 0): return True

        xstr = str(x)
        j = len(xstr)-1

        #floor div
        for i in range(0, len(xstr)//2) :
            if (xstr[i] == xstr[j]) :
                j -= 1
                continue
            else :
                return False
            
        return True
        

File: javascript/palindrome.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function (x) {
  if (x < -9) {
    return false;
  } else if (x < 9 && x >= 0) {
    return true;
  }

  let xstr = x.toString();
  let j = xstr.length - 1;

  //look from both ends
  for (let i = 0; i < Math.floor(xstr.length / 2); i++) {
    if (xstr.charAt(i) == xstr.charAt(j)) {
      j--;
      continue;
    } else {
      return false;
    }
  }
  return true;
};
